### PR TITLE
fix: Remove CVE-2023-45133 vulnerability

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR updates the versions of the packages **_nyc_** and _**@babel/traverse**_ to avoid the [CVE-2023-45133](https://security.snyk.io/vuln/SNYK-JS-BABELTRAVERSE-5962462) vulnerability.

## Specific Changes
  - Updated _**nyc**_ from 14.1.1 to 15.1.0 to use @babel/core ^7.23.2.
  - Added postinstall scripts to remove the _**@babel/traverse**_ packages that _orchestrator-core_ keeps.
  - Added **_@istanbuljs/nyc-config-typescript_** to to use as configuration for _nyc_ .

## Testing
The following image shows the _botframework-cli_ pipeline working after the changes.
image

### CODE REVIEW

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
The changes look good. Please add more details to the PR's description to understand why we are applying these changes.

File: packages/cli/.nycrc [hunk 1]
```diff
@@ -19,6 +19,5 @@
   ],
   "all": true,
   "cache": true,
-  "extends": "@istanbuljs/nyc-config-typescript",
```
Comment (sw-joelmut) [inline packages/cli/.nycrc]: Is there are reason why we are removing this extends? nyc recommends using this (https://github.com/istanbuljs/nyc?tab=readme-ov-file#typescript-projects)

**Comment by JhontSouth** [comment]
@ceciliaavila @sw-joelmut 
The feedback has already been applied

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted /1365

**Pull request closed without merging**